bionic/datatypes.py

```diff
@@ -72,15 +69,33 @@ class Result(object):
     Represents one value for one entity.
     '''
     def __init__(
-            self, query, value, file_path=None):
+            self, query, value, value_hash=None, file_path=None):
         self.query = query
         self.value = value
         self.file_path = file_path
+        # (For now) Only valid when value should persist.
```

Comment:
- Oh yes, this applies to `file_path` too. I'll fix that. 
- I put for now since I thought we will replace it with another decorator later (hashable / digestible). I can either remove it or expand on it. 
- `present` sounds better to me too.